cpp/src/db/db_object.cu

```diff
@@ -265,14 +246,6 @@ db_result<idx_t>& db_result<idx_t>::operator =(db_result<idx_t>&& other) {
 
 template<typename idx_t>
 db_result<idx_t>::~db_result() {
-  deleteData();
-}
```

Comment:
If you initialize dataValid and columnSize on definition, no need to have your custom default constructor, destructor, copy & move constructor & operators.